Runtime/Scripts/BaseMeshSync.cs

```diff
@@ -1041,9 +1043,8 @@ public abstract partial class BaseMeshSync : MonoBehaviour, IObservable<MeshSync
         if (dst == null || dst.material == null)
             return;
         
-        if (importerSettings.OverwriteExportedMaterials)
-            dst.ShouldApplyMaterialData = true;
-
+        dst.ShouldApplyMaterialData = isNewMaterial || importerSettings.OverwriteExportedMaterials;
```

Comment:
I am thinking we should probably make `ShouldApplyMaterialData` to a local variable along with this fix, seeing that we only use it inside this function.